src/main/kotlin/com/strumenta/kolasu/model/Processing.kt

```diff
@@ -272,12 +272,133 @@ fun Node.mapTree(operation: (Node) -> Node): Node {
 }
 
 /**
- * Replace this node with "other" (by modifying the children of the parent node.)
- * For this to work, assignParents() must have been called.
+ * Replace [this] node with [other] (by modifying the children of the parent node.)
+ * For this to work, [Node.assignParents] must have been called.
  */
 fun Node.replaceWith(other: Node) {
     if (this.parent == null) {
         throw IllegalStateException("Parent not set")
     }
     this.parent!!.transformTree { if (it == this) other else it }
 }
+
+/**
+ * Looks for [oldNode] in the lists of nodes in this node.
+ * When found, it is removed, and in its place the [newNodes] are inserted.
+ */
+fun Node.replaceWithSeveral(oldNode: Node, newNodes: List<Node>) {
+    findMutableListContainingChild(oldNode) { nodeList, index ->
+        nodeList.replaceWithSeveral(index, newNodes)
+        oldNode.parent = null
+        newNodes.forEach { node -> node.parent = this }
+    }
+}
+
+/**
+ * Looks for [targetNode] in the lists of nodes in this node.
+ * When found, it is removed.
+ */
+fun Node.remove(targetNode: Node) {
```

Comment:
But this does not remove the node if it is not in a list, right? Should we rename the method?

Reply:
Same as the other - or do you think the method name is too general?

Reply:
Yes, in this case the name is a bit general as it will not remove the node if it is not contained in a list, right?

Reply:
Done, is the name better?

Reply:
It is perfect, thank you!